src/api/providers/bedrock.ts

```diff
@@ -211,22 +208,24 @@ export class AwsBedrockHandler implements ApiHandler {
 	 * Creates a BedrockRuntimeClient with the appropriate credentials
 	 */
 	private async getBedrockClient(): Promise<BedrockRuntimeClient> {
-		const credentials = await this.getAwsCredentials()
-		let auth: any = {
-			credentials: {
-				accessKeyId: credentials.accessKeyId,
-				secretAccessKey: credentials.secretAccessKey,
-				sessionToken: credentials.sessionToken,
-			},
-		}
+		let auth: any
 
 		if (this.options.awsAuthentication === "apikey") {
 			auth = {
 				token: { token: this.options.awsBedrockApiKey },
 				authSchemePreference: ["httpBearerAuth"],
 			}
+		} else {
+			const credentials = await this.getAwsCredentials()
+			auth = {
+				credentials: {
+					accessKeyId: credentials.accessKeyId,
+					secretAccessKey: credentials.secretAccessKey,
+					sessionToken: credentials.sessionToken,
+				},
+			}
 		}
-
+		console.log({ auth })
```

Comment:
You wanna console logs the creds? This is a security violation.

Reply:
@massi-ang

Reply:
Maybe you can keep all the logs now for both of us to test and once I give you a written approval you can remove the logs and then I can give a proper approval and merge.

Reply:
well spotted. I removed it, but I will push it when you are done with the testing. If you still face the issue, please let me know if that objects contains the field "credentials" or "token".